src/LibraryViewExtension/LibraryViewController.cs

```diff
@@ -181,29 +181,60 @@ internal LibraryView AddLibraryView()
+            browser.LoadingStateChanged += LoadingStateChanged;
 
-            // TODO - this was added Fall 2017 due to issues with 
-            // library failing to load due to timing issues.  DYN-944 
-            // is a testing task that should make the final determination
-            // as it no longer seems required in CEF v65.0.1
-            /*
-            //wait for the browser to load before setting the resources
-            browser.LoadingStateChanged += (sender, args) =>
+            // Attempt to register required resources, but if the browser is still loading
+            // another attempt will be made when the loading state of the browser changes (see event above)
+            try
             {
-                //Wait for the Page to finish loading
-                if (args.IsLoading == false)
+                RegisterResources(browser);
+                string msg = "Successfully loaded library resources on first attempt.";
+                this.dynamoViewModel.Model.Logger.Log(msg);
+            }
+            catch(Exception ex)
+            {
+                string error = "The library browser is still loading, another attempt to load the resources will be made when loading is complete." +
+                    System.Environment.NewLine +
+                    "Exception: " + ex.Message;
+                this.dynamoViewModel.Model.Logger.LogError(error);
+            }
+
+            return view;
+        }
+
+        // This event is trigger twice,
+        // Once when loading is initiated either programmatically or by user action, 
+        // and once when loading is terminated due to completion, cancellation of failure.
+        private void LoadingStateChanged(object send, LoadingStateChangedEventArgs browserArgs)
+        {
+            // If the loading state has changed and the page has finished loading
+            if (!browserArgs.IsLoading)
+            {
+                // Attempt to load resources
+                try
                 {
                     RegisterResources(browser);
+                    string msg = "Successfully loaded library resources after a browser loading state change.";
+                    this.dynamoViewModel.Model.Logger.Log(msg);
                 }
-            };
-            */
-
-            return view;
+                catch(Exception ex)
+                {
+                    string error = "Failed to The library loaded library resources after a browser loading state change." +
+                        System.Environment.NewLine +
+                        "Exception: " + ex.Message;
+                    this.dynamoViewModel.Model.Logger.LogError(error);
+                }
+            }
+            // Browser is still loading or a browser failure has occured
+            else
+            {
+                string error = "The library browser is still loading or an error has occured, another attempt to load the resources will be made if loading can be completed.";
```

Comment:
@alfarok  Seems a second trial could always be made even if the first one succeed? Any downside?

Reply:
@QilongTang yes that is a good point.  I am still digging into the `LoadingStateChanged` behavior a bit.  In CEFSharp v57 we stopped manually calling `RegisterResources(browser)` and began to rely on this event.  After moving to v65 the library was never loading unless that code was restored to its original state (before we modified it last fall).  Unfortunately, they don't give you many properties to further determine why the state changed :confused: 

From what I can see it fires once on startup but the library is still loading so we do nothing.  It then fires again when loading is complete and we try to load the resources.  Ideally this should be all that is required.

Reply:
I think found a better solution digging through the docs! `browser.Loaded` does exactly what we need with less ambiguity compared to `browser.LoadingStateChanged`, testing it out now

Reply:
@QilongTang seems to be working and much cleaner solution

Reply:
Great. LGTM